bin/anthology/texmath.py

```diff
@@ -85,31 +85,31 @@ def _parse(self, everything, trg):
         """
         sxscript = False  # Tracks whether we're in a subscript/superscript
         for code in everything:
-            if isinstance(code, TexSoup.TexCmd):
+            if isinstance(code, TexCmd):
                 # code is a TeX command
                 self._parse_command(code, trg)
-            elif isinstance(code, TexSoup.TokenWithPosition):
+            elif isinstance(code, TexText) or isinstance(code, str):
                 # code is text
                 sxscript = self._parse_text(code, trg)
-            elif isinstance(code, TexSoup.Arg):
+            elif isinstance(code, TexGroup):
                 # If in subscript/superscript, wrap the entire element in respective tag
                 if sxscript:
                     my_trg = etree.Element(sxscript)
-                    self._parse(TexSoup.TexSoup(code.value).expr.everything, my_trg)
+                    self._parse(code.contents, my_trg)
                     trg.append(my_trg)
                     sxscript = False
                 # Otherwise, just parse it normally
                 else:
-                    self._parse(TexSoup.TexSoup(code.value).expr.everything, trg)
+                    self._parse(code.contents, trg)
             else:
                 log.error(f"TeX-math parser got unhandled element: {type(code)}")
 
     def _parse_command(self, code, trg):
-        args = list(code.arguments)
+        args = list(code.args)
         name = str(code.name)
         # TexSoup doesn't parse curly brackets correctly
```

Comment:
I've briefly tried it on the console and it seems that the newer TeXSoup _does_ parse curly brackets now:

```pycon
>>> x = TexSoup("some {T}est")
>>> x.expr.all
['some ', BraceGroup('T'), 'est']
```

So we can probably rewrite this to get rid of this hack (and the other one below). I can take a look at this!

Reply:
commit [b927fe5](https://github.com/acl-org/acl-anthology/pull/1843/commits/b927fe526843b9d11d1d516da177cbfc3d6a47e5) should have this fixed